datafusion/sql/src/utils.rs

```diff
@@ -119,9 +121,34 @@ pub(crate) fn check_columns_satisfy_exprs(
             _ => check_column_satisfies_expr(columns, e, message_prefix)?,
         }
     }
+    let column_names = columns
+        .iter()
+        .map(|c| format!("{}", c.schema_name()))
+        .collect::<Vec<_>>();
+
+    wildcard_fields.into_iter().try_for_each(|(table, field)| {
+        let column_name = qualified_name(table, field.name());
+        if !column_names.iter().any(|c| c == &column_name) {
```

Comment:
I think we need to ensure the column is either in group by and select statement. I'm confused why we check against wildcard_fields

Reply:
hmm ok. I added back the checking for the rules. I tried to add more comments to explain why we need those checks. 
About the error message, we could enhance it in the future 🤔 .